velox/substrait/TypeUtils.cpp

```diff
@@ -43,6 +43,31 @@ int64_t bytesOfType(const TypePtr& type) {
   }
 }
 
+void getTypesFromCompoundName(
+    const std::string& compoundName,
+    std::vector<std::string>& types) {
+  types.clear();
+  // Get the position of ":" in the compoundName.
+  std::size_t pos = compoundName.find(":");
+  // Get the parameter types.
+  std::string funcTypes;
+  if (pos == std::string::npos) {
+    return;
+  } else {
+    if (pos == compoundName.size() - 1) {
+      return;
+    }
+    funcTypes = compoundName.substr(pos + 1);
+  }
+  // Split the types with delimiter.
+  std::string delimiter = "_";
```

Comment:
Would you double check that this logic supports nested structs, e.g. ROW(BIGINT(), ROW(BIGINT(), DOUBLE());